goal_src/levels/village1/sage.gc

```diff
@@ -5,3 +5,1422 @@
+               (joint-control-channel-group-eval!
+                a0-13
+                (the-as art-joint-anim #f)
+                num-func-seek!
+                )
+               )
+              (ja-done? 0)
+              )
+       (if
+        (and
+         (-> self reminder-played)
+         (< 81920.0 (vector-vector-distance (-> self root trans) (camera-pos)))
+         )
+        (go-virtual idle)
+        )
+       )
+      )
+     )
+    (when (= (if (> (-> self skel active-channels) 0)
+              (-> self skel root-channel 0 frame-group)
+              )
+           (-> self draw art-group data 4)
+           )
+     (let ((a0-20 (-> self skel root-channel 0)))
+      (set!
+       (-> a0-20 frame-group)
+       (the-as art-joint-anim (-> self draw art-group data 5))
+       )
+      (set!
+       (-> a0-20 param 0)
+       (the
+        float
+        (+
+         (->
+          (the-as art-joint-anim (-> self draw art-group data 5))
+          data
+          0
+          length
+          )
+         -1
+         )
+        )
+       )
+      (set! (-> a0-20 param 1) 1.0)
+      (set! (-> a0-20 frame-num) 0.0)
+      (joint-control-channel-group!
+       a0-20
+       (the-as art-joint-anim (-> self draw art-group data 5))
+       num-func-seek!
+       )
+      )
+     (until (begin
+             (suspend)
+             (let ((a0-21 (-> self skel root-channel 0)))
+              (set!
+               (-> a0-21 param 0)
+               (the float (+ (-> a0-21 frame-group data 0 length) -1))
+               )
+              (set! (-> a0-21 param 1) 1.0)
+              (joint-control-channel-group-eval!
+               a0-21
+               (the-as art-joint-anim #f)
+               num-func-seek!
+               )
+              )
+             (ja-done? 0)
+             )
+      (empty)
+      )
+     )
+    )
+   (none)
+   )
+  )
+
+;; failed to figure out what this is:
+(defstate play-anim (sage)
+  :virtual #t
+  :exit
+  (behavior ()
+   (set! (-> self draw bounds w) 10240.0)
+   (let ((a0-1 (handle->process (-> self assistant))))
+    (if a0-1
+     (deactivate a0-1)
+     )
+    )
+   ((-> (method-of-type process-taskable play-anim) exit))
+   (case (get-task-status (game-task intro)) 
+    (((task-status need-resolution))
+      (clear-pending-settings-from-process
+       *setting-control*
+       self
+       'music-volume-movie
+       )
+      (close-specific-task! (game-task intro) (task-status need-resolution))
+      (start 'play (get-continue-by-name *game-info* "game-start"))
+      )
+    )
+   (none)
+   )
+  :trans
+  (behavior ()
+   (case (get-task-status (game-task intro)) 
+    (((task-status need-reward-speech))
+      (spool-push *art-control* "sage-intro-sequence-d2" 0 self -1.0)
+      )
+    )
+   ((-> (method-of-type process-taskable play-anim) trans))
+   (none)
+   )
+  )
+
+;; definition for method 39 of type sage
+(defmethod target-close? sage ((obj sage))
```

Comment:
Looks like this method might actually be something like `should-display?` or `is-visible?` or something like that.